newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -117,6 +121,7 @@ export default class AuthenticatedUserProvider extends React.Component<
     this.props.authentication.addUserLogoutListener(
       this._fetchUserProfileWithoutThrowingErrors
     );
+    this.props.authentication.addUserLogoutListener(onUserLogout);
```

Comment:
You could avoid this and call `onUserLogout` (rename it to `identifyLoggedOutUserForAnalytics` to be consistent with the rest?) inside `_fetchUserProfile` (when `if (!firebaseUser) {` is true) and inside `_resetAuthenticatedUser`.... but maybe what you did is simpler. So let's keep it.

Note: we should probably have a "componentWillUnmount" which would call `removeEventListener`.

Reply:
we should probably indeed.
But as it's working I'm quite afraid to touch this whole provider logic!

Reply:
Let's not waste more time on this as it's already working.